Content.Shared/_CM14/Scoping/SharedScopeSystem.cs

```diff
@@ -0,0 +1,193 @@
+using Robust.Shared.Timing;
+
+namespace Content.Shared._CM14.Scoping;
+
+public abstract partial class SharedScopeSystem : EntitySystem
+{
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+    [Dependency] private readonly ActionContainerSystem _actionContainer = default!;
+    [Dependency] private readonly SharedTransformSystem _transformSystem = default!;
+    [Dependency] private readonly SharedHandsSystem _handsSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly INetManager _net = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        InitializeUser();
+        SubscribeLocalEvent<ScopeComponent, GotEquippedHandEvent>(OnEquip);
+        SubscribeLocalEvent<ScopeComponent, GotUnequippedHandEvent>(OnUnequip);
+        SubscribeLocalEvent<ScopeComponent, DroppedEvent>(OnDrop);
+        SubscribeLocalEvent<ScopeComponent, HandDeselectedEvent>(OnDeselectHand);
+
+        SubscribeLocalEvent<ScopeComponent, GetItemActionsEvent>(OnGetActions);
+        SubscribeLocalEvent<ScopeComponent, ToggleActionEvent>(OnToggleAction);
+
+        SubscribeLocalEvent<ScopeComponent, ComponentShutdown>(OnShutdown);
+
+        SubscribeLocalEvent<ScopeComponent, MapInitEvent>(OnMapInit);
+    }
+
+    private void OnMapInit(EntityUid uid, ScopeComponent component, MapInitEvent args)
+    {
+        _actionContainer.EnsureAction(uid, ref component.ScopingToggleActionEntity, component.ScopingToggleAction);
+        Dirty(uid, component);
+    }
+
+    private void OnEquip(EntityUid uid, ScopeComponent component, GotEquippedHandEvent args)
+    {
+        component.User = args.User;
+        Dirty(uid, component);
+
+        if(_net.IsServer)
+            EnsureComp<ScopeUserComponent>(args.User);
+    }
+
+    private void OnUnequip(EntityUid uid, ScopeComponent component, GotUnequippedHandEvent args)
+    {
+        StopScopingHelper(uid, component, args.User);
+    }
+
+    private void OnDrop(EntityUid uid, ScopeComponent component, DroppedEvent args)
+    {
+        StopScopingHelper(uid, component, args.User);
+    }
+
+    private void OnDeselectHand(EntityUid uid, ScopeComponent component, HandDeselectedEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (component.IsScoping)
+            StopScoping(uid, component, args.User);
+    }
+
+    private void OnGetActions(EntityUid uid, ScopeComponent component, GetItemActionsEvent args)
+    {
+        args.AddAction(ref component.ScopingToggleActionEntity, component.ScopingToggleAction);
+    }
+
+    private void OnToggleAction(EntityUid uid, ScopeComponent component, ToggleActionEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (!(_handsSystem.TryGetActiveItem(args.Performer, out var heldItem) && heldItem == uid))
```

Comment:
This check can be simplified so that it's easier to understand
```suggestion
        if (!_handsSystem.TryGetActiveItem(args.Performer, out var heldItem) || heldItem != uid)
```